Hestia/app/src/main/java/hestia/backend/Activator.java

```diff
@@ -3,94 +3,83 @@
 /**
  * This class represents a single activator on a device. A single device can have multiple activators.
  * The activator has an id so we can reference it on the server.
- * Furthermore there is a string name and a string state type,
- * which is currently used for casting the generic state variable to an actual type.
+ * Furthermore there is a string name and a string state name,
+ * which is currently used for casting the generic state variable to an actual name.
  * @see Device
  * @see ActivatorState
  */
 public class Activator {
-    private int id;
+    private String activatorId;
+    private int rank;
     private ActivatorState state;
     private String name;
 
     /**
      * Creates an Activator with the specified id, state and name.
-     * @param id the id of the activator
+     * @param activatorId the id of the activator
      * @param state the current state of the activator, this is a wrapper around a generic type
      * @param name the name of the activator
      */
-    public Activator(int id, ActivatorState state, String name) {
-        this.id = id;
+    public Activator(String activatorId, Integer rank, ActivatorState state, String name) {
+        this.activatorId = activatorId;
+        this.rank = rank;
         this.state = state;
         this.name = name;
     }
 
-    /**
-     * Returns the id of the activator.
-     * @return the id of the activator
-     */
-    public int getId() {
-        return id;
+    public String getId() {
+        return activatorId;
     }
 
-    /**
-     * Replaces the current id of the activator with the specified one.
-     * @param id the new id of the activator
-     */
-    public void setId(int id) {
-        this.id = id;
+
+    public void setId(String activatorId) {
+        this.activatorId = activatorId;
+    }
+
+    public Integer getRank() {
+        return rank;
+    }
+
+    public void setRank(Integer rank) {
+        this.rank = rank;
     }
 
-    /**
-     * Returns the current state of the activator.
-     * @return the current state of the activator
-     */
     public ActivatorState getState() {
         return state;
     }
 
-    /**
-     * Replaces the current state of the activator with the specified one.
-     * @param state the new state of the activator
-     */
     public void setState(ActivatorState state) {
         this.state = state;
     }
 
-    /**
-     * Returns the name of the activator.
-     * @return the name of the activator
-     */
     public String getName() {
         return name;
     }
 
-    /**
-     * Replaces the current name of the activator with the specified one.
-     * @param name the new name of the activator
-     */
     public void setName(String name) {
         this.name = name;
     }
 
     @Override
     public boolean equals(Object o) {
         if (this == o) return true;
-        if (o == null || getClass() != o.getClass()) return false;
+        if (!(o instanceof Activator)) return false;
```

Comment:
instanceof is technically bad OOP (should be handled by inheritance and strongly typed design), but acceptable if there _is no other way_ to do it better. Look into it.